src/test/kotlin/edu/udo/asyncjobqueue/AsyncJobQueueTest.kt

```diff
@@ -18,4 +21,386 @@ class AsyncJobQueueTest {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+        val job = Runnable {}
+
+        // when:
+        jobQueue.submit(job)
+
+        // then:
+        assertThat(executor.submitted).hasSize(1)
+    }
+
+    @Test
+    fun `When a job is submitted then it is executed`() {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+        val job = mock(Runnable::class.java)
+        jobQueue.submit(job)
+
+        // when:
+        executor.runNext()
+
+        // then:
+        assertThat(executor.submitted).isEmpty()
+        verify(job).run()
+    }
+
+    @Test
+    fun `When a job is submitted while another is running then the job is not submitted for execution`() {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+
+        val runningJob = Runnable {}
+        jobQueue.submit(runningJob)
+
+        val job = Runnable {}
+
+        // when:
+        jobQueue.submit(job)
+
+        // then:
+        assertThat(executor.submitted).hasSize(1)
+    }
+
+    @Test
+    fun `When a job is submitted after another completed then the job is submitted for execution`() {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+
+        val completedJob = Runnable {}
+        jobQueue.submit(completedJob)
+        executor.runNext()
+
+        val job = Runnable {}
+
+        // when:
+        jobQueue.submit(job)
+
+        // then:
+        assertThat(executor.submitted).hasSize(1)
+    }
+
+    @Test
+    fun `When a running job completes then the next job is submitted for execution`() {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+
+        val completingJob = mock(Runnable::class.java)
+        jobQueue.submit(completingJob)
+        val job = Runnable {}
+        jobQueue.submit(job)
+
+        assertThat(executor.submitted).hasSize(1)
+
+        // when:
+        executor.runNext()
+
+        // then:
+        assertThat(executor.submitted).hasSize(1)
+        verify(completingJob).run()
+    }
+
+    @Test
+    fun `When three jobs are submitted then they are executed in order`() {
+        // given:
+        val executor = ManualExecutorService()
+        val jobQueue = AsyncJobQueue.create(executor)
+
+        val job1 = mock(Runnable::class.java)
+        jobQueue.submit(job1)
+        val job2 = mock(Runnable::class.java)
+        jobQueue.submit(job2)
+        val job3 = mock(Runnable::class.java)
+        jobQueue.submit(job3)
+
+        // when:
+        executor.runNext()
+        executor.runNext()
+        executor.runNext()
+
+        // then:
+        assertThat(executor.submitted).isEmpty()
```

Comment:
These assertions do not guarantee that the jobs are executed in order as stated in the test name.

Reply:
Fixed this.